### PR TITLE
feat: add basic trial details endpoint consumption [DET-3640]

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

needed for https://github.com/determined-ai/determined/pull/867

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/services/decoder.ts [hunk 1]
```diff
@@ -181,6 +183,14 @@ const ioTrialToTrial = (io: ioTypeTrialSummary): TrialSummary => {
   };
 };
 
+export const jsonToTrialDetails = (data: unknown): TrialDetails => {
+  const io = decode<ioTypeTrialDetails>(ioTrialDetails, data);
+  return { ...io,
```
Comment (hkang1) [inline webui/react/src/services/decoder.ts:188]: **suggestion:** I saw this pattern in the experiment detail endpoint and ran into duplicate fields wherever field names changed to camelcase due to the spread operator. For example, under experiment config, you would find both `checkpoint_storage` and `checkpointStorage` defined. It's a bit more labor but probably better to explicitly set each field without the spread operator.

**Comment by hkang1** [comment]
**question:** When looking at json responses, what's the easiest way figure out what all the fields are and what their expect values and value types are? I've used a combination of looking at the decoders in Elm and looking through Go code to do this but it's not straight forward...

File: webui/react/src/services/decoder.ts (hunk 1 above)
Comment (hamidzr) [reply webui/react/src/services/decoder.ts:188]: I agree I think at some point we need to write a dynamic converter that does this. we need the reverse for going back from a parsed experiment config to one that the server understands. for example when forking experiments we need to send back `checkpoint_storage` and not the camelcase version.

**Comment by hamidzr** [comment]
> **question:** When looking at json responses, what's the easiest way figure out what all the fields are and what their expect values and value types are? I've used a combination of looking at the decoders in Elm and looking through Go code to do this but it's not straight forward...

i do a combination of go code, elm decoders, and API response.  this is because we don't have documentation and the endpoint responses are not typed. with  the new API we solve both of those: (/swagger-ui) and proto files.

**Pull request merged**